gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/rest/api/service/v4/impl/ApiSearchService_SearchIdsTest.java

```diff
@@ -0,0 +1,219 @@
+    public static void cleanSecurityContextHolder() {
+        // reset authentication to avoid side effect during test executions.
+        SecurityContextHolder.setContext(
+            new SecurityContext() {
+                @Override
+                public Authentication getAuthentication() {
+                    return null;
+                }
+
+                @Override
+                public void setAuthentication(Authentication authentication) {}
+            }
+        );
+    }
+
+    @Before
+    public void setUp() {
+        ApiMapper apiMapper = new ApiMapper(
+            new ObjectMapper(),
+            planService,
+            flowService,
+            parameterService,
+            workflowService,
+            new CategoryMapper(categoryService)
+        );
+        apiSearchService =
+            new ApiSearchServiceImpl(
+                apiRepository,
+                apiMapper,
+                new GenericApiMapper(apiMapper, new ApiConverter()),
+                primaryOwnerService,
+                categoryService,
+                searchEngineService,
+                apiAuthorizationService
+            );
+    }
+
+    @Test
+    public void should_return_empty_page_if_no_results() {
+        QueryBuilder<GenericApiEntity> apiEntityQueryBuilder = QueryBuilder.create(GenericApiEntity.class).setQuery("*").setSort(null);
+        var ids = List.of("id-1", "id-2");
+
+        var filters = new HashMap<String, Object>();
+        filters.put("api", ids);
+        apiEntityQueryBuilder.setFilters(filters);
+
+        when(searchEngineService.search(eq(GraviteeContext.getExecutionContext()), eq(apiEntityQueryBuilder.build())))
+            .thenReturn(new SearchResult(List.of()));
+
+        final var apis = apiSearchService.searchIds(GraviteeContext.getExecutionContext(), "*", filters, null, false);
+
+        assertThat(apis).isNotNull();
+        assertThat(apis.size()).isEqualTo(0);
+    }
+
+    @Test
+    public void should_return_api_ids() {
+        QueryBuilder<GenericApiEntity> apiEntityQueryBuilder = QueryBuilder.create(GenericApiEntity.class).setQuery("*").setSort(null);
+        var ids = List.of("id-1", "id-2");
+
+        var filters = new HashMap<String, Object>();
+        filters.put("api", ids);
+        apiEntityQueryBuilder.setFilters(filters);
+
+        when(searchEngineService.search(eq(GraviteeContext.getExecutionContext()), eq(apiEntityQueryBuilder.build())))
+            .thenReturn(new SearchResult(List.of("api-id")));
+
+        final var apis = apiSearchService.searchIds(GraviteeContext.getExecutionContext(), "*", filters, null, false);
+
+        assertThat(apis).isNotNull();
+        assertThat(apis.size()).isEqualTo(1);
+        assertThat(apis).isEqualTo(List.of("api-id"));
+    }
+
+    @Test
+    public void should_not_add_empty_query() {
+        QueryBuilder<GenericApiEntity> apiEntityQueryBuilder = QueryBuilder.create(GenericApiEntity.class).setQuery(null).setSort(null);
```

Comment:
Should it be?
```suggestion
        QueryBuilder<GenericApiEntity> apiEntityQueryBuilder = QueryBuilder.create(GenericApiEntity.class).setQuery("").setSort(null);
```
or just
```suggestion
        QueryBuilder<GenericApiEntity> apiEntityQueryBuilder = QueryBuilder.create(GenericApiEntity.class).setSort(null);
```

Reply:
The expected built query should have a `null` query attribute. So either `.setQuery(null)` or a missing `setQuery` would be the equivalent.